### PR TITLE
Demo improvements

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by schmmd** [comment]
@joelgrus can you take a look and give me a round of feedback?  This is WIP.

File: allennlp/commands/serve.py [hunk 1]
```diff
@@ -9,10 +13,17 @@ def add_subparser(parser: argparse._SubParsersAction) -> argparse.ArgumentParser
 
     subparser.add_argument('--port', type=int, default=8000)
     subparser.add_argument('--workers', type=int, default=1)
+    subparser.add_argument('--configfile', type=argparse.FileType('r'), default=None, help="path to a JSON file specifying the configuration for the models")
```
Comment (schmmd) [inline allennlp/commands/serve.py]: How do I read this setting if I name it `config-file`?

File: allennlp/service/predictors/bidaf.py [hunk 2]
```diff
@@ -14,4 +14,4 @@ def predict_json(self, inputs: JsonDict) -> JsonDict:
         passage = TextField(self.tokenizer.tokenize(passage_text) + [SquadReader.STOP_TOKEN],
                             token_indexers=self.token_indexers)
 
-        return sanitize(self.model.predict_span(question, passage))
+        return sanitize(self.model.predict_span(question, passage)["best_span_str"])
```
Comment (schmmd) [inline allennlp/service/predictors/bidaf.py]: I will pass the full data and filter to only this field in the web application.

File: allennlp/service/predictors/semantic_role_labeler.py [hunk 3]
```diff
@@ -16,24 +16,51 @@ def __init__(self, model: Model,
 
         self.nlp = spacy.load('en', parser=False, vectors=False, entity=False)
 
+    def make_srl_string(words: List[str], tags: List[str]) -> str:
```
Comment (schmmd) [inline allennlp/service/predictors/semantic_role_labeler.py]: Suggestions appreciated--I'm not sure the best way to code this up in python.

File: allennlp/commands/serve.py (hunk 1 above)
Comment (joelgrus) [reply allennlp/commands/serve.py]: behind the scenes it should get saved as `args.config_file`

File: allennlp/service/server_sanic.py [hunk 4]
```diff
@@ -3,21 +3,21 @@
...
 from sanic import Sanic, response, request
 from sanic.exceptions import ServerError
+from typing import Dict
 
-DEFAULT_ARCHIVE_FILES = {
-        'machine-comprehension': 'tests/fixtures/bidaf/serialization/model.tar.gz',
-        'semantic-role-labeling': 'tests/fixtures/srl/serialization/model.tar.gz',
+default_config = {
+        'machine-comprehension': 'https://s3-us-west-2.amazonaws.com/allennlp/models/bidaf-model-2017.08.26.tar.gz',
+        'semantic-role-labeling': 'https://s3-us-west-2.amazonaws.com/allennlp/models/srl-model-2017.08.28.tar.gz',
```
Comment (joelgrus) [inline allennlp/service/server_sanic.py]: I don't know if you care, but I think you also do urls like

`https://allennlp.s3.amazonaws.com/models/...`

File: allennlp/service/predictors/semantic_role_labeler.py (hunk 3 above)
Comment (joelgrus) [reply allennlp/service/predictors/semantic_role_labeler.py]: no obviously better way jumps out at me

File: allennlp/service/server_sanic.py [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
 from sanic.exceptions import ServerError
 
-DEFAULT_ARCHIVE_FILES = {
-        'machine-comprehension': 'tests/fixtures/bidaf/serialization/model.tar.gz',
-        'semantic-role-labeling': 'tests/fixtures/srl/serialization/model.tar.gz',
-        'textual-entailment': 'tests/fixtures/decomposable_attention/serialization/model.tar.gz'
+from allennlp.models.archival import load_archive
+from allennlp.service.predictors import Predictor
+
+DEFAULT_CONFIG = {
```
Comment (joelgrus) [inline allennlp/service/server_sanic.py]: I feel like it's weird for `DEFAULT_CONFIG` to live here but not be used in this file.

IMO, either

* `DEFAULT_CONFIG` should instead live in `serve.py`, or
* the logic for "if config not specified, then use DEFAULT_CONFIG" should be here

I don't have a strong opinion as to which of these is better

**Pull request merged**